Element84/filmdrop-ui
# Ba/feature/add vitest setup

**Related Issue(s):**

- NA

**Proposed Changes:**

1. add vitest base setup and config
2. add example test for app.jsx
3. change pre-commit hooks to run test command with --run (same as --watchall=false but for vitest)

> **_NOTE:_**  Currently API fetch calls are nested inside functions created in components which will make testing much more complicated since we'd have to mock those just to get a render to correctly load. I left some example tests to check for renders as a pattern, but made comments for related refactor to separate service calls from presentational logic. 

**PR Checklist:**

- [X ] I have added my changes to the [CHANGELOG](https://github.com/Element84/filmdrop-ui/blob/main/CHANGELOG.md) **or** a CHANGELOG entry is not required.
